viewing_party/main.py

```diff
@@ -0,0 +1,240 @@
+        x = movie["genre"]
+        if x not in genre_map:
+            genre_map[x] = 1
+        else:
+            genre_map[x] += 1
+    
+    most_watched_genre = (max(genre_map, key=genre_map.get)) if genre_map else None
+
+    return most_watched_genre 
+
+    # current_max = 0
+    # for k, v in genre_map.items():
+    #     if v > current_max:
+    #         current_max = v
+    
+    # for k, v in genre_map.items():
+    #     if v == current_max:
+    #         return k
+
+# Wave 3
+'''
+Determine which movies the user has watched, 
+but none of their friends have watched.
+'''
+def get_unique_watched(user_data):
+
+    mine_watched = {}
+    for movie in user_data["watched"]:
+        mine_watched[movie["title"]] = True
+    
+    friends_watched = {}
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched[movie["title"]] = True
+
+    diff = []
+    for title in mine_watched.keys():
+        if title not in friends_watched:
+            diff.append({"title": title})
+    return diff
+
+    # movie_map = {}
+    # for k,v in user_data.items():
+    #     if k == "watched":
+    #         for movie in v:
+    #             x = movie["title"]
+    #             if x not in movie_map:
+    #                 movie_map[x] = 0  
+    #     if k == "friends":
+    #         for friend in v:
+    #             for v in friend.values():
+    #                 for title in v:
+    #                     y = title["title"]
+    #                     if y in movie_map:
+    #                         movie_map[y] += 1
+
+    # # Return a list of dictionaries, that represents a list of movies
+    # holder = []
+    # for k,v in movie_map.items():
+    #     if v == 0:
+    #         holder.append(k)
+
+    # result = [{"title": i} for i in holder]
+
+    # return result
+
+'''
+Determine which movies at least one of the user's friends have watched, 
+but the user has not watched.
+'''
+def get_friends_unique_watched(user_data):
+
+    mine_watched = {}
+    for movie in user_data["watched"]:
+        mine_watched[movie["title"]] = True
+    
+    friends_watched = {}
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched[movie["title"]] = True
+
+    diff = []
+    for title in friends_watched.keys():
+        if title not in mine_watched:
+            diff.append({"title": title})
+    return diff  
+
+# Wave 4
+'''
+Determine a list of recommended movies. Add to list if:
+The user has not watched it
+At least one of the user's friends has watched
+The "host" of the movie is a service that is in the user's "subscriptions"
+'''
+def get_available_recs(user_data):
+
+    friends_watched = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_watched.append(movie)
+    
+    result = []
+    for movie in friends_watched:
+        if movie["host"] in user_data["subscriptions"]:
```

Comment:
Another way to eliminate duplicates is to check to see if the movie is already in the result set and only add it if it not:
```
if movie["host"] in user_data["subscriptions"] and movie not in result:
```